lmfdb/cluster_pictures/web_cluster_picture.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+from sage.all import cached_function
+
+from lmfdb import db
+from lmfdb.logger import make_logger
+wcplog = make_logger("WCP")
+
+@cached_function
+def cp_display_knowl(label, name=None, img=None):
+    if not name:
+        name = "Cluster Picture %s" % label
+    if not img:
+        img = name
+    return '<a title = "%s [clusterpicture.data]" knowl="clusterpicture.data" kwargs="label=%s">%s</a>' % (name, label, img)
+
+def cp_knowl_guts(label):
+    out = ''
+    wcp = WebClusterPicture(label)
+    if wcp.is_null():
+        return 'Cannot find cluster picture %s' % label
+    out += "<b>Cluster picture %s</b>" % label
+    out += '<div>'
+    out += '<img src="'
+    out += wcp.image()
+    out += '" alt="cluster_picture_image"/>'
+    out += '<br>Depth: '
+    out += str(wcp.depth())
+    out += '<br>Size: '
+    out += str(wcp.size())
+    out += '<br>Potential toric rank: '
+    out += str(wcp.potential_toric_rank())
+    out += '<br>Potential good reduction curve: '
+    out += str(wcp.potential_good_reduction())
+    out += '<br>Potential good reduction Jacobian: '
+    out += str(wcp.potential_good_jacobian_reduction())
+    out += '</div>'
+    #out += '<div align="right">'
```

Comment:
These commented lines should be removed, or edited if you intend to eventually have cluster picture homepages and want to keep a placeholder.

Reply:
Yeah, I think this is why I kept them, because I don't know what the final state will be. But I can also remove them, if you prefer.